### PR TITLE
Migrate Task to Spring Cloud Build 2.0.x

### LINKED ISSUE #313: Spike: Migrate Spring Cloud Task to Spring Cloud Build 2.0.x

### PR DESCRIPTION
**Pull request by cppwfs**
resolves #313

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -137,10 +137,10 @@
...
 	<properties>
-		<spring-cloud-stream.version>1.2.1.RELEASE</spring-cloud-stream.version>
+		<spring-cloud-stream.version>2.0.0.BUILD-SNAPSHOT</spring-cloud-stream.version>
 		<spring-cloud-deployer-spi.version>1.2.0.RELEASE</spring-cloud-deployer-spi.version>
 		<spring-cloud-deployer-local.version>1.2.0.RELEASE</spring-cloud-deployer-local.version>
-		<spring-cloud-stream-binder-rabbit.version>1.2.0.RELEASE</spring-cloud-stream-binder-rabbit.version>
+		<spring-cloud-stream-binder-rabbit.version>2.0.0.BUILD-SNAPSHOT</spring-cloud-stream-binder-rabbit.version>
 		<spring-cloud-deployer-resource-support.version>1.2.0.RELEASE</spring-cloud-deployer-resource-support.version>
 		<spring-cloud-deployer-resource-maven.version>1.2.0.RELEASE</spring-cloud-deployer-resource-maven.version>
 		<spring-batch.version>3.0.7.RELEASE</spring-batch.version>
```
Comment (mminella) [inline pom.xml]: This should be Batch 4 for Boot 2.

File: spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/TaskBatchExecutionListenerTests.java [hunk 2]
```diff
@@ -144,38 +144,38 @@ public void testAutobuiltDataSourceNoJob() {
...
 		TaskExplorer taskExplorer = this.applicationContext.getBean(TaskExplorer.class);
 
-		Page<TaskExecution> page = taskExplorer.findTaskExecutionsByName("application", new PageRequest(0, 1));
+		Page<TaskExecution> page = taskExplorer.findTaskExecutionsByName("application", PageRequest.of(0, 1));
 
 		Set<Long> jobExecutionIds = taskExplorer.getJobExecutionIdsByTaskExecutionId(page.iterator().next().getExecutionId());
 
 		assertEquals(1, jobExecutionIds.size());
-		assertEquals(0, (long) taskExplorer.getTaskExecutionIdByJobExecutionId(jobExecutionIds.iterator().next()));
+		assertEquals(1, (long) taskExplorer.getTaskExecutionIdByJobExecutionId(jobExecutionIds.iterator().next()));
```
Comment (mminella) [inline spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/TaskBatchExecutionListenerTests.java]: Why did the value here change?

File: spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/TaskBatchExecutionListenerTests.java [hunk 3]
```diff
@@ -144,38 +144,38 @@ public void testAutobuiltDataSourceNoJob() {
...
+		Page<TaskExecution> page = taskExplorer.findTaskExecutionsByName("application", PageRequest.of(0, 1));
 
 		Set<Long> jobExecutionIds = taskExplorer.getJobExecutionIdsByTaskExecutionId(page.iterator().next().getExecutionId());
 
 		assertEquals(2, jobExecutionIds.size());
 		Iterator<Long> jobExecutionIdsIterator = jobExecutionIds.iterator();
-		assertEquals(0, (long) taskExplorer.getTaskExecutionIdByJobExecutionId(jobExecutionIdsIterator.next()));
-		assertEquals(0, (long) taskExplorer.getTaskExecutionIdByJobExecutionId(jobExecutionIdsIterator.next()));
+		assertEquals(1, (long) taskExplorer.getTaskExecutionIdByJobExecutionId(jobExecutionIdsIterator.next()));
+		assertEquals(1, (long) taskExplorer.getTaskExecutionIdByJobExecutionId(jobExecutionIdsIterator.next()));
```
Comment (mminella) [inline spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/TaskBatchExecutionListenerTests.java]: Same comment as above about the value changing...

Comment (cppwfs) [inline spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/TaskBatchExecutionListenerTests.java]: By adding the EmbeddedDataSourceConfiguration to the context it sets the initial id start to 1.

File: spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/TaskBatchExecutionListenerTests.java (hunk 2 above)
Comment (cppwfs) [inline spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/TaskBatchExecutionListenerTests.java]: By adding the EmbeddedDataSourceConfiguration to the context it sets the initial id start to 1.

File: pom.xml (hunk 1 above)
Comment (cppwfs) [inline pom.xml]: okeedokee.

**Comment by mminella** [comment]
Merged with additional polish as d95f46002fb5828b02e054adaea1e7808dc35f21 and 2c2712ada4a8a624db16e2e48333724c7532d082.

**Pull request closed without merging**